homeassistant/components/cast/__init__.py

```diff
@@ -49,9 +52,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Cast from a config entry."""
     await home_assistant_cast.async_setup_ha_cast(hass, entry)
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    hass.data[DOMAIN] = {}
+    await async_process_integration_platforms(hass, DOMAIN, _process_cast_platform)
```

Comment:
This should either be done in async_setup, but I think it's better to just do it on demand whenever media_browse is called for the first time in the media player.

Reply:
We should also define a `Protocol` object with expected methods. https://github.com/home-assistant/core/blob/dev/homeassistant/components/diagnostics/__init__.py#L49

Reply:
`Protocol` class defined in 10ffd80